### PR TITLE
feat: deprecated trace config

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes [EMQX-9474](https://emqx.atlassian.net/browse/EMQX-9474)

### <samp>🤖 Generated by Copilot at 07538dd</samp>

This pull request adds the `payload_encode` option to the HTTP API for creating traces, which allows the user to control the format of the payload in the trace file. It also deprecates the `trace` and `trace.payload_encode` configurations in the `emqx_schema.erl` file, and handles the error case when the trace creation fails due to a rolling upgrade. It updates the `emqx_trace` module and the related test cases to support the new option and the new trace record format.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [x] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [x] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx/test/emqx_trace_SUITE.erl [hunk 1]
```diff
@@ -24,7 +24,9 @@
 -include_lib("emqx/include/emqx.hrl").
 -include_lib("snabbkaffe/include/snabbkaffe.hrl").
 
--record(emqx_trace, {name, type, filter, enable = true, start_at, end_at}).
+-record(emqx_trace, {
```
Comment (sstrigler) [inline apps/emqx/test/emqx_trace_SUITE.erl:27]: Why not

```erlang
-include(“emqx_trace.hrl”).
```

**Review by sstrigler — APPROVED** [review]
just minor things, lgtm

File: apps/emqx/src/emqx_trace/emqx_trace.erl [hunk 2]
```diff
@@ -59,7 +59,8 @@
 -ifdef(TEST).
 -export([
     log_file/2,
-    find_closest_time/2
+    find_closest_time/2,
+    migrate_trace/0
```
[the remark this answers is not in the document]
Comment (zhongwencool) [reply apps/emqx/src/emqx_trace/emqx_trace.erl]: Good catch!

File: apps/emqx/test/emqx_trace_SUITE.erl (hunk 1 above)
Comment (zhongwencool) [reply apps/emqx/test/emqx_trace_SUITE.erl]: Done

File: apps/emqx_management/src/emqx_mgmt_api_trace.erl [hunk 3]
```diff
@@ -265,6 +266,20 @@ fields(trace) ->
...
+                    ""
+                    "Determine the format of the payload format in the trace file.<br/>\n"
+                    "`text`: Text-based protocol or plain text protocol.\n"
+                    " It is recommended when payload is JSON encoded.<br/>\n"
+                    "`hex`: Binary hexadecimal encode."
+                    "It is recommended when payload is a custom binary protocol.<br/>\n"
+                    "`hidden`: payload is obfuscated as `******`"
+                    "",
+                default => text,
+                required => false
```
Comment (thalesmg) [inline apps/emqx_management/src/emqx_mgmt_api_trace.erl:281]: Only one of those is required:

```suggestion
                default => text
```

File: apps/emqx_management/src/emqx_mgmt_api_trace.erl [hunk 4]
```diff
@@ -421,6 +436,11 @@ trace(post, #{body := Param}) ->
                 code => 'DUPLICATE_CONDITION',
                 message => ?TO_BIN([Name, " Duplication Condition"])
             }};
+        {error, {bad_type, _}} ->
+            {409, #{
+                code => 'BAD_TYPE',
+                message => <<"Rolling upgrade in progress, create failed">>
```
Comment (thalesmg) [inline apps/emqx_management/src/emqx_mgmt_api_trace.erl:442]: Is the message correct?  It seems it would be something more like "Bad encoding type"?

**Review by thalesmg — APPROVED** [review]

**Review by thalesmg — APPROVED** [review]

**Pull request merged**